app/src/androidTest/java/ch/epfl/sdp/peakar/user/OtherAccountTest.java

```diff
@@ -0,0 +1,119 @@
+package ch.epfl.sdp.peakar.user;
+
+import android.net.Uri;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.google.android.gms.auth.api.Auth;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.auth.FirebaseAuth;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+import ch.epfl.sdp.peakar.database.Database;
+import ch.epfl.sdp.peakar.user.services.Account;
+import ch.epfl.sdp.peakar.user.services.AuthAccount;
+import ch.epfl.sdp.peakar.user.services.AuthService;
+import ch.epfl.sdp.peakar.user.services.OtherAccount;
+import ch.epfl.sdp.peakar.user.services.providers.firebase.FirebaseAuthService;
+
+import static ch.epfl.sdp.peakar.TestingConstants.BASIC_USERNAME;
+import static ch.epfl.sdp.peakar.user.AuthAccountTest.registerAuthUser;
+import static ch.epfl.sdp.peakar.user.AuthAccountTest.removeAuthUser;
+
+public class OtherAccountTest {
+    private static String user2;
+
+    /* Set up the environment */
+    @BeforeClass
+    public static void init() {
+        /* Make sure no user is signed in before tests */
+        AuthService.getInstance().signOut(InstrumentationRegistry.getInstrumentation().getTargetContext());
+
+
+        /* Create a new one */
+        registerAuthUser();
+
+        user2 = (BASIC_USERNAME + AuthService.getInstance().getID()).substring(0, AuthAccount.NAME_MAX_LENGTH - 2);
+    }
+
+    /* Clean environment */
+    @AfterClass
+    public static void end() {
+        removeTestUsers();
+        removeAuthUser();
+    }
+
+    /* Make sure that an account is signed in and as new before each test */
+    @Before
+    public void createTestUser() {
```

Comment:
I see that this exist in several tests regarding DB. Should it be moved to some kind of FireBaseTestHelper class?

Reply:
You're right. For now I modified the method `registerAuthUser()` so the rest of the code inside the createTestUser is not repeated.
I see the point in adding that class, but, for now, I think we use only that and the remove methods (that are defined in `AuthAccountTest` class), do you think I should create the tester class for these two methods?